cli/src/opts/multi_wallet.rs

```diff
@@ -287,12 +289,18 @@ impl MultiWallet {
     }
 
     pub fn mnemonics(&self) -> Result<Option<Vec<LocalWallet>>> {
-        if let (Some(mnemonic_paths), Some(mnemonic_indexes)) =
-            (self.mnemonic_paths.as_ref(), self.mnemonic_indexes.as_ref())
-        {
+        if let Some(mnemonic_paths) = self.mnemonic_paths.as_ref() {
             let mut wallets = vec![];
-            for (path, mnemonic_index) in mnemonic_paths.iter().zip(mnemonic_indexes) {
-                wallets.push(self.get_from_mnemonic(path, *mnemonic_index)?)
+            let hd_paths: Vec<Option<&str>> = match &self.hd_paths {
+                None => (0..mnemonic_paths.len()).map(|_| None).collect(),
+                Some(paths) => paths.into_iter().map(|p| Some(p.as_ref())).collect(),
+            };
```

Comment:
let's do a `if let` here and use `std::iter::repeat`

```suggestion
let hd_paths:Vec<_> = if let Some(ref hd_paths) = self.hd_paths {
                hd_paths.into_iter().map(String::as_str).map(Some).collect()
            } else {
                std::iter::repeat(None).take(mnemonic_paths.len()).collect()
            };
```